cogito/putter.go

```diff
@@ -134,22 +135,40 @@ func (putter *ProdPutter) ProcessInputDir() error {
 	return nil
 }
 
-func (putter *ProdPutter) Sinks() []Sinker {
-	return []Sinker{
-		GitHubCommitStatusSink{
+func (putter *ProdPutter) Sinks() ([]Sinker, error) {
+	var err error
+	supportedSinks := map[string]Sinker{
+		"github": GitHubCommitStatusSink{
 			Log:     putter.log.Named("ghCommitStatus"),
 			GhAPI:   putter.ghAPI,
 			GitRef:  putter.gitRef,
 			Request: putter.Request,
 		},
-		GoogleChatSink{
+		"gchat": GoogleChatSink{
 			Log: putter.log.Named("gChat"),
 			// TODO putter.InputDir itself should be of type fs.FS.
 			InputDir: os.DirFS(putter.InputDir),
 			GitRef:   putter.gitRef,
 			Request:  putter.Request,
 		},
 	}
+	sinksParams := putter.Request.Params.Sinks
+	if len(sinksParams) == 0 {
+		// No sink specified, it will return all supported sinks.
+		sinksParams = make([]string, 0, len(supportedSinks))
+		for k := range supportedSinks {
+			sinksParams = append(sinksParams, k)
+		}
+	}
+	sinkers := make([]Sinker, 0, len(sinksParams))
+	for _, s := range sinksParams {
+		sinkers = append(sinkers, supportedSinks[s])
+	}
+
+	if len(sinkers) == 0 {
```

Comment:
This could only be true if it did not blow up above with unknown sinks. In that case, you could as well have encountered a config like `sinks: [github, slack]` and so `sinkers` would hold only the `github` sink and you would silently ignore the `slack` sink. Or it blew up above and that test is useless because you ensure that `sinkParams` is not empty.

Reply:
Good point, thanks

Reply:
This should be addressed in https://github.com/Pix4D/cogito/pull/108/commits/0984de9d8541a7b939fea573f891aaf4c998c457